### PR TITLE
Authorization > Check canCreate flag to toggle create button visibility (companies/sites)

### LINKED ISSUE #2332: Authorization > Check canCreate flag to toggle create button visibility (companies/sites)
Example: the admin user can create a payment method for the Billing but only for himself in his user's profile but cannot create one for the other users. In that case we have to hide the Create button. This enhancement will further improve the authorization's granularity.

This issue covers companies and sites.

### PR DESCRIPTION
**Pull request by Fonsu**
Closes #2332

### CODE REVIEW

File: src/app/types/Table.ts [hunk 1]
```diff
@@ -120,6 +120,7 @@ export interface TableActionDef {
   icon?: string;
   color?: ButtonColor;
   disabled?: boolean;
+  hidden?: boolean;
```
Comment (LucasBrazi06) [inline src/app/types/Table.ts:123]: Visible?

File: src/app/shared/table/table.component.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <!-- Create Left Actions -->
         <ng-container *ngFor="let actionDef of dataSource.tableActionsDef; let i=index">
           <!-- Action Button -->
-          <button mat-raised-button *ngIf="actionDef.type === 'button' && actionDef.id !== 'reset_filters'"
+          <button mat-raised-button *ngIf="actionDef.type === 'button' && actionDef.id !== 'reset_filters'" [hidden]="actionDef.hidden"
```
Comment (LucasBrazi06) [inline src/app/shared/table/table.component.html:10]: visible?

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/app/pages/organization/sites/list/sites-list-table-data-source.ts [hunk 3]
```diff
@@ -80,6 +82,9 @@ export class SitesListTableDataSource extends TableDataSource<Site> {
       // Get Sites
       this.centralServerService.getSites(this.buildFilterValues(),
         this.getPaging(), this.getSorting()).subscribe((sites) => {
+        if (sites.canCreate) {
+          this.createAction.hidden = false;
```
[the remark this answers is not in the document]
Comment (Fonsu) [reply src/app/pages/organization/sites/list/sites-list-table-data-source.ts:86]: Do you mean I should use  something like`this.createAction.visible = !!sites.canCreate;`?

If you mean the check is not needed, I'm confused. Isn't it a valid scenario that a user can list sites but not create them?

File: src/app/pages/organization/companies/list/companies-list-table-data-source.ts [hunk 4]
```diff
@@ -205,6 +203,7 @@ export class CompaniesListTableDataSource extends TableDataSource<Company> {
         if (actionDef.action) {
           (actionDef as TableCreateCompanyActionDef).action(CompanyDialogComponent, this.dialog, this.refreshData.bind(this));
         }
+        actionDef.visible = false;
```
Comment (LucasBrazi06) [inline src/app/pages/organization/companies/list/companies-list-table-data-source.ts:206]: visible is false by default, no need to set it to false here.

File: src/app/pages/organization/sites/list/sites-list-table-data-source.ts [hunk 5]
```diff
@@ -233,6 +236,7 @@ export class SitesListTableDataSource extends TableDataSource<Site> {
         if (actionDef.action) {
           (actionDef as TableCreateSiteActionDef).action(SiteDialogComponent, this.dialog, this.refreshData.bind(this));
         }
+        actionDef.visible = false;
```
Comment (LucasBrazi06) [inline src/app/pages/organization/sites/list/sites-list-table-data-source.ts:239]: visible is false by default, no need to set it to false here.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

**Review by LucasBrazi06 — APPROVED** [review]
GUT!

**Pull request merged**